metricshub-ping-extension/src/main/java/org/sentrysoftware/metricshub/extension/ping/PingRequestExecutor.java

```diff
@@ -0,0 +1,49 @@
+package org.sentrysoftware.metricshub.extension.ping;
+
+/*-
+ * ╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲
+ * MetricsHub Ping Extension
+ * ჻჻჻჻჻჻
+ * Copyright 2023 - 2024 Sentry Software
+ * ჻჻჻჻჻჻
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * ╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱
+ */
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import org.sentrysoftware.metricshub.engine.strategy.utils.RetryOperation;
+
+public class PingRequestExecutor {
+
+	Boolean ping(String hostname, int maxAttempts, int timeout) throws UnknownHostException {
+		InetAddress pingAddress = InetAddress.getByName(hostname);
+
+		return RetryOperation
+			.<Boolean>builder()

```

Comment:
I'm wondering whether this `RetryOperation` is required, as `isReachable(timeout)` keeps sending `echo` ICMP packets every second until either 1 reply is received, or the specified timeout is reached.

If I understand correctly, `isReachable(5000)` will perform:

* 0:00:00: ECHO
* 0:00:01: ECHO
* 0:00:02: ECHO
* 0:00:03: ECHO
* **RESPONSE FOR FIRST ECHO FINALLY RECEIVED**
* Return `true`

Or:

* 0:00:00: ECHO
* 0:00:01: ECHO
* 0:00:02: ECHO
* 0:00:04: ECHO
* 0:00:05: ECHO
* **TIMEOUT**
* Return `false`

So there would be no need to perform any retry. The use just need to increase the timeout.